hydra_pywr/runner.py

```diff
@@ -629,16 +669,37 @@ def generate_array_recorder_resource_scenarios(self):
                     new_col_names.append(colname.split(':')[1].strip())
                 df.columns = new_col_names
 
+            if "__:" in recorder.name:
+                nodename = recorder.name.split(":")[0].replace("__", "")
+                attrname = recorder.name.split(":")[1]
+            else:
+                nodename = "network"
+                attrname = recorder.name
+
+            filename = f'{attrname}.h5'
+            resultstore = self.resultstores.get(filename)
+            if resultstore is None:
+                resultstore = pandas.HDFStore(os.path.join(self.results_location, filename), mode='w')
+                self.resultstores[filename] = resultstore
+
+            resultstore.put(f"{nodename}", df)
+            resultstore[f"{nodename}"].attrs['pandas_type'] = 'frame'
+
             # Convert to JSON for saving in hydra
-            value = df.to_json(date_format='iso', date_unit='s')
+            value = json.dumps({
+                "data":
+                {
+                    "url": f"s3://{self.bucket_name}/{self.s3_path}/{attrname}.h5",
```

Comment:
As those strings are being combined as parts of a url, is there a possibility of some characters needing encoding?

`urllib` has `urlencode` and `urlunsplit` for safe url building.

Reply:
These aren't a 'proper' url in the sense that, for example urlencode produces a key like "bucket=test" which is not appropriate for the creation of the s3 path.